haas/api.py

```diff
@@ -423,6 +428,42 @@ def node_detach_network(node, nic, network):
     return '', 202
 
 
+@rest_call('PUT', '/node/<node>/metadata/<label>', Schema({
+    'node': basestring, 'label': basestring, 'value': object,
+}))
+def node_set_metadata(node, label, value):
+    """Register metadata on a node.
+
+    If the label already exists, the value will be updated.
+    """
+    get_auth_backend().require_admin()
+    node = _must_find(model.Node, node)
+    obj_inner = _namespaced_query(node, model.Metadata, label)
+    if obj_inner is not None:
+        metadata = _must_find_n(node, model.Metadata, label)
+        metadata.value = json.dumps(value)
```

Comment:
Is there a reason we aren't accepting value as a `basestring` in the first place?

Is json.dumps() here as a means of escaping the input?